mobly/controllers/attenuator.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3.4
+#
+# Copyright 2016 Google Inc.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controller module for attenuators.
+
+Sample Config:
+"Attenuator": [
+    {
+        "Address": "192.168.1.12",
+        "Port": 23,
+        "Model": "minicircuits",
+        "PathCount": 4,
+        "Paths": ["AP1-2G", "AP1-5G", "AP2-2G", "AP2-5G"]
+    },
+    {
+        "Address": "192.168.1.14",
+        "Port": 23,
+        "Model": "minicircuits",
+        "PathCount": 1,
+        "Paths": ["AP-DUT"]
+    }
+]
+"""
+import importlib
+import logging
+
+MOBLY_CONTROLLER_CONFIG_NAME = "Attenuator"
+KEY_ADDRESS = "Address"
+KEY_PORT = "Port"
+
+
+def create(configs):
+    objs = []
+    for config in configs:
+        attn_model = config["Model"]
+        # Import the correct driver module for the attenuator device
+        module_name = "mobly.controllers.attenuator_lib.%s" % attn_model
+        module = importlib.import_module(module_name)
+        # Create each
+        path_cnt = config["PathCount"]
+        attn_device = module.AttenuatorDevice(path_cnt)
+        attn_device.model = attn_model
+        instances = attn_device.open(config[KEY_ADDRESS], config[KEY_PORT])
```

Comment:
Yes it is.
We can easily change this later if we need to support a different driver. And no test code needs to change if someone chooses to try running with a different hardware.
The point is to avoid test scripts having to change because hardware changed.